### PR TITLE
added an example workflow to the documentation

### PR DESCRIPTION
**Pull request by thsetz**

### CODE REVIEW

File: docs-source/sample_workflow.rst [hunk 1]
```diff
@@ -0,0 +1,258 @@
...
+
+.. code-block:: python
+
+  dst = os.path.join("/tmp",name, "wb")
+  with open(dst, "wb") as fp:
+    fp.write(content)
+
+  print(f"report was written to file {dst}.")
+
+Output
```
Comment (deveaud-m) [inline docs-source/sample_workflow.rst:238]: This section can be deleted.

**Review by deveaud-m — CHANGES_REQUESTED** [review]
Thanks for this contribution.

I made some comments and suggestions.

File: docs-source/sample_workflow.rst [hunk 2]
```diff
@@ -0,0 +1,258 @@
...
+        TokenScope.WRITE,
+    )
+    with open(".token", "w") as fp:
+        fp.write(token)
+    print("token written to .token")
+  else:
+    print("Reuse token from .token")
+  # Load the token
+  with open(".token", "r") as fp:
+    token = fp.read()
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:57]: OK

File: docs-source/sample_workflow.rst [hunk 3]
```diff
@@ -0,0 +1,258 @@
...
+
+
+Login to the fossology Server
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:60]: OK

File: docs-source/sample_workflow.rst [hunk 4]
```diff
@@ -0,0 +1,258 @@
...
+        test_folder = folder
+
+  if not folder_already_created:
+    print("Folder needs creation")
+    test_folder = foss.create_folder(
+        foss.rootFolder, folder_name, description=folder_desc
+    )
+    print(f"Created {test_folder.name} with description {test_folder.description}")
+  assert test_folder.name == folder_name
+  assert test_folder.description == folder_desc
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:95]: OK

File: docs-source/sample_workflow.rst [hunk 5]
```diff
@@ -0,0 +1,258 @@
...
+  if not group_already_created:
+    # XXX create_group does not return created group
+    foss.create_group(group_name)
+    for group in foss.list_groups():
+        if group.name == group_name:
+            test_group = group
+    print(f"Created group named {test_group.name}")
+
+  assert test_group
+  assert test_group.name == group_name
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:124]: If the group already exists, create_group throws an exception:

- fossology.exceptions.FossologyApiError: ('Group clearing already exists, failed to create group or no group name provided', <Response [500]>)

That is, why i check it again - and use the assert .

File: docs-source/sample_workflow.rst [hunk 6]
```diff
@@ -0,0 +1,258 @@
...
+    FOSSOLOGY_SERVER = "http://fossology/repo"
+
+Create JWT Token
+================
+
+Create token - if not already done - and store it in local file for reuse.
+
+.. code-block:: python
+
+  if not os.path.isfile(".token"):
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst]: For consistency  I used pathlib instead.

File: docs-source/sample_workflow.rst [hunk 7]
```diff
@@ -0,0 +1,258 @@
...
+
+Upload File 
+===========
+We first get an example file from our github repository testenvironment and then
+upload it to the server. 
+
+.. code-block:: python
+
+  filename = "my_base-files_11.tar.xz"
+  if not os.path.isfile(os.path.join("/tmp", filename)):
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:137]: OK

File: docs-source/sample_workflow.rst [hunk 8]
```diff
@@ -0,0 +1,258 @@
...
+    url = "https://github.com/fossology/fossology-python/blob/master/tests/files/base-files_11.tar.xz"
+    print(f"Get a local copy of {filename} from {url}")
+    r = requests.get(url)
+    with open(os.path.join("/tmp", filename), "wb") as fp:
+        fp.write(r.content)
+  assert os.path.isfile(os.path.join("/tmp", filename))
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:143]: OK

File: docs-source/sample_workflow.rst [hunk 9]
```diff
@@ -0,0 +1,258 @@
...
+  group_already_created = False
+
+  for group in foss.list_groups():
+    if group.name == group_name:
+        group_already_created = True
+        print(f"Group {group_name} already created")
+        test_group = group
+
+  if not group_already_created:
+    # XXX create_group does not return created group
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:116]: OK

File: docs-source/sample_workflow.rst [hunk 10]
```diff
@@ -0,0 +1,258 @@
...
+        file=test_file_path,
+        description="Test upload via fossology-python lib",
+        group=test_group.name,
+        access_level=AccessLevel.PUBLIC,
+    )
+  assert my_upload.uploadname == test_file
+
+Start default scan jobs
+=======================
+
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:172]: OK

File: docs-source/sample_workflow.rst [hunk 11]
```diff
@@ -0,0 +1,258 @@
...
+  print(f"report created with id {report_id} ")
+
+Download report
+===============
+
+.. code-block:: python
+
+  content, name = foss.download_report(report_id, test_group.name)
+
+Write report to disk
```
[the remark this answers is not in the document]
Comment (thsetz) [reply docs-source/sample_workflow.rst:227]: OK

File: docs-source/sample_workflow.rst (hunk 1 above)
Comment (thsetz) [reply docs-source/sample_workflow.rst:238]: OK

File: docs-source/sample_workflow.rst (hunk 2 above)
[the remark this answers is not in the document]
Comment (deveaud-m) [reply docs-source/sample_workflow.rst]: @thsetz you can resolve the discussions once you pushed the requested changes as a new commit.

File: docs-source/sample_workflow.rst (hunk 3 above)
[the remark this answers is not in the document]
Comment (deveaud-m) [reply docs-source/sample_workflow.rst]: @thsetz if you commit the suggestion the discussion will be resolved automatically

File: docs-source/sample_workflow.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (deveaud-m) [reply docs-source/sample_workflow.rst]: We should catch the exception in this example to show users of the API how to handle the different cases: group is created or group already exists.

**Comment by deveaud-m** [comment]
Thanks for addressing all the discussions. Only one comment regarding group creation handling is left.

**Comment by thsetz** [comment]
Maybe it's a good idea to rewrite sample_workflow.rst with doctest. I just tried it - but that. collides with some doctests in the library - which would need to be fixed too.

File: docs-source/sample_workflow.rst (hunk 5 above)
[the remark this answers is not in the document]
Comment (deveaud-m) [reply docs-source/sample_workflow.rst]: Discussion has been addressed.

**Review by deveaud-m — APPROVED** [review]
All changes look good. Ready to be merged.

**Pull request merged**